aten/src/ATen/native/vulkan/ops/BinaryElementwise.cpp

```diff
@@ -0,0 +1,432 @@
+    } else {
+      TORCH_CHECK(false, "Not implemented!");
+    }
+  }
+  command_pool.submit(context->gpu().queue, command_buffer);
+
+  return convert(v_output);
+}
+
+Tensor& binary_elementwise_scalar_(
+    Tensor& self,
+    const Scalar other,
+    const c10::optional<Scalar> alpha_arg,
+    const api::Shader::Descriptor& shader_descriptor) {
+  api::Context* const context = api::context();
+
+  TORCH_CHECK(
+      self.is_vulkan(),
+      "Vulkan: In-place add is only supported on Vulkan tensors.");
+
+  vTensor& v_self = convert(self);
+
+  api::Command::Pool& command_pool = context->command().pool;
+  api::Command::Buffer& command_buffer = command_pool.stream();
+  {
+    if C10_LIKELY (v_self.has_image()) {
+      const float other_val = alpha_arg
+          ? other.to<float>() * alpha_arg->to<float>()
+          : other.to<float>();
+      const struct Block final {
+        uvec3 extents;
+        float other;
+      } block{
+          v_self.extents(),
+          other_val,
+      };
+
+      context->dispatch(
+          command_buffer,
+          {
+              VK_DESCRIPTOR_TYPE_STORAGE_IMAGE,
+              VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER,
+          },
+          shader_descriptor,
+          v_self.extents(),
+          context->gpu().adapter->local_work_group_size(),
+          // Read-Write access triggers an async synchronization if necessory
+          // and inserts appropriate barriers if hazards are detected.
+          v_self.image(
+              command_buffer,
+              vTensor::Stage::Compute,
+              vTensor::Access::Read | vTensor::Access::Write),
+          // Object lifetime is managed by the resource pool.
+          // It is OK not to keep track of the handle.
+          context->resource().pool.uniform(block).object);
+    } else {
+      TORCH_CHECK(false, "Not implemented!");
+    }
+  }
+  command_pool.submit(context->gpu().queue, command_buffer);
+
+  return self;
+}
+
+Tensor binary_elementwise_tensor(
+    const Tensor& self_arg,
+    const Tensor& other_arg,
+    const c10::optional<Scalar> alpha_arg,
+    const api::Shader::Descriptor& shader_descriptor) {
+  check_inputs(self_arg, other_arg);
+  api::Context* const context = api::context();
+
+  const Tensor self = self_arg.is_vulkan() ? self_arg : self_arg.vulkan();
+  const vTensor& v_self = convert(self);
+
+  const Tensor other = other_arg.is_vulkan() ? other_arg : other_arg.vulkan();
+  const vTensor& v_other = convert(other);
+
+  vTensor v_output{
+      context,
+      broadcast_first_input(v_self, v_other) ? v_other.sizes() : v_self.sizes(),
+      v_self.options(),
+  };
+
+  api::Command::Pool& command_pool = context->command().pool;
+  api::Command::Buffer& command_buffer = command_pool.stream();
+  {
+    if C10_LIKELY (v_self.has_image() && v_other.has_image()) {
+      const float alpha = alpha_arg ? alpha_arg->to<float>() : 1.0;
+      const struct Block final {
+        uvec3 extents;
+        uint32_t fill_0;
```

Comment:
If you decide to pack tighter you can replace this with float alpha.  Completely up to you.  It's not going to be a performance boost and only a very minor performance saving.